app/domain/authentication/authn_k8s/application_identity.rb

```diff
@@ -112,7 +112,9 @@ def constraint_from_id constraint_name
       end
 
       def host_id_suffix
-        @host_id_suffix ||= @host_id.split('/').last(3)
+        # We first split the host id by ':' and take the last part to remove the '<account>:host:' prefix.
+        # Leaving it in a one-liner for memoization.
+        @host_id_suffix ||= (@host_id.split(':')[2]).split('/').last(3)
```

Comment:
Consider extracting a method. ExtractSuffixFromHostId ?
Can we have cases where hostId does not include ':' ?

Reply:
created a new method

Reply:
The host id will always include `account:kind:identifier`. It is built here: `host_id = "#{@account}:" + @common_name.k8s_host_name.sub('host/', 'host:')`